src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject.cs

```diff
@@ -23,39 +21,24 @@ internal sealed partial class CPSProject : AbstractProject
             IVsHierarchy hierarchy,
             string language,
             Guid projectGuid,
-            string commandLineForOptions,
+            string binOutputPath,
             IServiceProvider serviceProvider,
             VisualStudioWorkspaceImpl visualStudioWorkspaceOpt,
             HostDiagnosticUpdateSource hostDiagnosticUpdateSourceOpt,
             ICommandLineParserService commandLineParserServiceOpt)
             : base(projectTracker, reportExternalErrorCreatorOpt, projectDisplayName, projectFilePath,
                    hierarchy, language, projectGuid, serviceProvider, visualStudioWorkspaceOpt, hostDiagnosticUpdateSourceOpt, commandLineParserServiceOpt)
         {
-            // Initialize the options.
-            SetCommandLineArguments(commandLineForOptions);
-
             // We need to ensure that the bin output path for the project has been initialized before we hookup the project with the project tracker.
-            // If we were unable to set the output path from SetCommandLineArguments (due to null output file name or directory in the given commandLineForOptions),
-            // we set a default unique output path.
-            if (this.TryGetBinOutputPath() == null)
-            {
-                var uniqueDefaultOutputPath = PathUtilities.CombinePathsUnchecked(Path.GetTempPath(), projectDisplayName + projectGuid.GetHashCode().ToString());
-                SetOutputPathAndRelatedData(objOutputPath: uniqueDefaultOutputPath, hasSameBinAndObjOutputPaths: true);
-            }
-
-            Contract.ThrowIfNull(this.TryGetBinOutputPath());
+            // Also initialize the objOutputPath to the same value.
+            SetOutputPathAndRelatedData(objOutputPath: binOutputPath, binOutputPath: binOutputPath);
```

Comment:
This looks wrong - why are you setting them to the same path? if you need 'obj' force me to pass it to you.

This state of this object is very unclear - we set to obj to bin, only to set it again to the real 'obj'?